src/transactionsPage/utils.ts

```diff
@@ -0,0 +1,120 @@
+import { Transaction, Filters } from "./";
+import { SelectOptions } from "./filter";
+import { StatementStatistics } from "src/util/appStats";
+
+export const getAppNames = (transactions: Transaction[]) => {
+  return transactions.reduce(
+    (acc, current) => {
+      const twin = acc.some(
+        (item: SelectOptions) => item.label === current.stats_data.app,
+      );
+      return twin
+        ? acc
+        : acc.concat([
+            {
+              label: current.stats_data.app,
+              value: current.stats_data.app,
+            },
+          ]);
+    },
+    [{ label: "All", value: "All" }],
+  );
+};
+
+export const addTransactionStatements = (
+  transactions: Transaction[],
+  statements: StatementStatistics[],
+) => {
+  const data = transactions.map((item: Transaction) => {
```

Comment:
When using map/reduce etc. use a less generic variable as the parameter. `t` or `tx` would work better in this case because otherwise it's difficult to guess what `item` represents later in the code.

Similar to the function above, you can make this easier to read and understand. Break the actions up into sequential steps instead of nested ones:

array of transactions -> map to array of array of statement IDs -> map to array of array of statement queries -> map to array of strings

if you want to merge the `transactionStatements` object back into the Transaction you can either do that after you're done with the processing or map over the `transactions` and do the processing per-transaction in a separate function.